dali/pipeline/operators/op_spec.h

```diff
@@ -32,6 +33,40 @@
 
 namespace dali {
 
+namespace detail {
+
+template <typename T, typename S>
+void copy_vector(std::vector<T> &out, const std::vector<S> &in) {
+  out.reserve(in.size());
+  out.clear();
+  for (decltype(auto) v : in) {
+    out.emplace_back(v);
+  }
+}
+
+/** @brief This overload simply forwards the reference */
+template <typename T>
+std::vector<T> &&convert_vector(std::vector<T> &&v) {
+  return std::move(v);
+}
+
+/** @brief This overload simply forwards the reference */
+template <typename T>
+const std::vector<T> &convert_vector(const std::vector<T> &v) {
+  return v;
+}
+
+/** @brief This overload converts elements from v and returns a vector of converted objects */
+template <typename T, typename S>
+std::enable_if_t<!std::is_same<T, S>::value, std::vector<T>>
+convert_vector(const std::vector<S> &v) {
+  std::vector<T> out;
+  copy_vector(out, std::move(v));
```

Comment:
Why do you move a `const&`?
```suggestion
  copy_vector(out, v);
```